Swift/Adhan/Adhan.swift

```diff
@@ -403,6 +408,35 @@ public struct Qibla {
     }
 }
 
+/* Sunnah times for a location and date using the given prayer times.
+All prayer times are in UTC and should be display using an DateFormatter that
+has the correct timezone set. */
+public struct SunnahTimes {
+    public let midnight: Date
+    public let lastThird: Date
+
+    public init?(from prayerTimes: PrayerTimes) {
+        let cal: Calendar = .utcGregorian
+        let tomorrow = cal.date(byAdding: .day, value: 1, to: prayerTimes.fajr)!
+        
+        guard let nextDayPrayers = PrayerTimes(
+            coordinates: prayerTimes.coordinates,
+            date: cal.dateComponents([.year, .month, .day], from: tomorrow),
+            calculationParameters: prayerTimes.calculationParameters) else {
+                // unable to determine tomorrow prayer times
+                return nil
+        }
+        
+        let minutes = cal.dateComponents([.minute], from: prayerTimes.maghrib, to: nextDayPrayers.fajr).minute!
```

Comment:
It is more efficient, and technically more accurate to use timeInterval and do the calculation in seconds. We can then do roundedMinute() at the very end.